packages/commcare/src/Adaptor.js

```diff
@@ -77,23 +81,48 @@ export function get(path, params = {}, callback = s => s) {
       params
     );
 
+    let allItems = [];
+    let next;
+    let nextState = state;
+    let offset = resolvedParams.offset;
+    const limit = resolvedParams.limit;
+
     try {
-      const response = await request(
-        state.configuration,
-        `/a/${domain}/api/v0.5/${resolvedPath}`,
-        {
-          method: 'GET',
-          params: resolvedParams,
-          contentType: 'application/json',
+      do {
+        // fetch a page
+
+        const response = await request(
+          state.configuration,
+          `/a/${domain}/api/v0.5/${resolvedPath}`,
+          {
+            method: 'GET',
+            params: {
+              ...resolvedParams,
+              offset: offset,
+              limit: limit,
+            },
+            contentType: 'application/json',
+          }
+        );
+        nextState = prepareNextState(state, response, callback);
+
+        if (Array.isArray(nextState.data)) {
```

Comment:
I am a bit uncomfortable with this.

`get()` is generic. Whatever the server sends to us, we should write to `state.data`.

If we get a single case, state .data should not be an array. But this logic here forces it into an array.

Basically the `allItems` array isn't necessarily the answer here. We only want `allItems` if there's pagination.